### PR TITLE
feat(file-tree): make ignored directories configurable

### LINKED ISSUE #1015: File tree hard-ignores dirs named build/dist/target/… by name — respect .gitignore or make it configurable
### Summary
The Files view hard-filters any directory whose **name** exactly matches a fixed set (`build`, `dist`, `node_modules`, `target`, `.next`, `.gradle`, …), regardless of whether it is actually build output. A directory legitimately named e.g. `build` (a docs category, a source folder) is therefore invisible in the file tree — even when it is git-tracked and **not** in `.gitignore`.

### Where
`server/index.js`:

```js
const IGNORED_DIRS = new Set(['node_modules', 'dist', 'build', /* … */]);
entries.filter(e => !(e.isDirectory() && IGNORED_DIRS.has(e.name)));
```

Introduced/widened in #669 (`perf(file-tree): … widen default ignore list`).

### Impact
No override exists: the list is hardcoded, there is no setting/env, and it does not consult `.gitignore`. Any project with a legitimately-named `build`/`dist`/… folder (e.g. `docs/build/`) cannot be browsed or opened in the web UI.

### Repro
1. Create a git-tracked folder `docs/build/foo.md` that is **not** in `.gitignore`.
2. Open the Files view → the `build/` folder is missing from the tree.

### Suggested fix (any one)
1. Respect `.gitignore` as the source of truth (only hide what git actually ignores).
2. Make `IGNORED_DIRS` configurable (a setting or env var).
3. At minimum, do not hide directories that are git-tracked, so real source/docs folders named `build` still show.

### Version
`@cloudcli-ai/cloudcli` v1.36.1

### PR DESCRIPTION
**Pull request by HandyS11**
Closes #1015

## Problem

The file tree hid a fixed set of directory names with no override (`server/modules/file-tree/file-tree.service.ts`). That cut both ways:

- A directory legitimately named `build` or `dist` was invisible even when git-tracked (the original report in #1015).
- Generated output that was *not* on the list could not be hidden. On a .NET solution, `bin/` and `obj/` accounted for **13,138 of 14,408** entries, pushing the tree past the 10,000-entry cap so the Files tab failed with `Project file tree exceeds the 10,000 entry limit. Choose a narrower project directory or add ignore rules.` — advice that was impossible to follow, since the list was hardcoded.

## Change

- `IGNORED_DIRECTORY_NAMES` becomes `DEFAULT_IGNORED_DIRECTORY_NAMES`, with `bin` and `obj` added.
- The effective list is injected into `createFileTreeService` through a new `ignoredDirectories` gateway and resolved per walk, so an updated list applies to the next request without a restart.
- The list is persisted in the existing `app_config` table under `file_tree_ignored_directories`; an absent or unreadable row means "never configured" and falls back to the defaults.
- `GET`/`PUT /api/file-tree/settings/ignored-directories` expose it. Submitted values are trimmed, deduplicated, and rejected if they contain a path separator, so the stored list stays a name-by-name comparison and can never be read as a path pattern.
- New **Settings → Files** tab to edit the list, with a reset to defaults. Saving dispatches an event so an open file tree refetches instead of showing a stale tree.
- Strings added for all 11 locales.

A project with a tracked `docs/build/` can now simply remove `build` from its list, which resolves #1015 without hiding real source folders by name.

## Testing

- `npm test` — 272 pass. 10 new tests cover the defaults (including `bin`/`obj`), a custom list replacing the defaults, the settings read/write service methods, input normalization and rejection, and the two routes.
- `npm run typecheck`, `npm run lint`, `npm run build` — clean (no new lint findings).
- Verified end-to-end against a running server on the .NET solution above: the tree that previously returned the 10,000-entry error now loads with **1,259** entries, `PUT` normalizes `["  bin ","obj","bin",""]` to `["bin","obj"]`, and `["src/generated"]` is rejected with 400.

## Notes

- This PR deliberately does not change `respectGitignore`. Worth flagging separately: the Files view never sends it (`src/utils/api.js:258`), so the entry cap is enforced against a tree filtered only by the name list; only the @-mention picker requests gitignore filtering.

## Summary by CodeRabbit

- **New Features**
  - Added a Files settings tab for managing directories hidden from the file tree.
  - Users can add, remove, save, and reset ignored directory names.
  - File trees refresh automatically after settings are saved.
  - Added default exclusions for common build directories.
  - Added validation and clear error handling for invalid entries.
- **Localization**
  - Added translated Files settings text across supported languages.

### CODE REVIEW

File: src/components/settings/view/tabs/FilesSettingsTab.tsx [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+            />
+            <Button variant="outline" onClick={addDirectory} disabled={!newDirectory.trim()}>
+              {t('filesSettings.add')}
+            </Button>
+          </div>
+
+          <div className="flex flex-wrap items-center gap-2">
+            <Button onClick={() => void save(ignoredDirectories)} disabled={isSaving}>
+              {isSaving ? t('filesSettings.saving') : t('filesSettings.save')}
+            </Button>
```
[the remark this answers is not in the document]
Comment (HandyS11) [reply src/components/settings/view/tabs/FilesSettingsTab.tsx:179]: Confirmed and fixed in 66ef360. On a failed load the tab now renders only the error: it tracks whether settings were ever read and returns before the editor, so there is no Save or Reset button to press. Verified by stubbing the GET to 500 in a browser — the panel shows the error text with 0 Save and 0 Reset buttons, where previously both were live over an empty list.

File: server/shared/types.ts [hunk 2]
```diff
@@ -1020,10 +1020,23 @@ export type FileTreeLogger = {
  * and logging are all explicit so service construction has no hidden process,
```
[the remark this answers is not in the document]
Comment (HandyS11) [reply server/shared/types.ts]: Good catch, fixed in 66ef360. The gateway type and its doc comment now sit above the `FileTreeServiceDependencies` doc block, so each comment is attached to its own type again. I also extended that block to mention the new ignored-directory dependency.

File: server/modules/file-tree/file-tree.service.ts [hunk 3]
```diff
@@ -11,13 +11,19 @@ import type {
 } from '@/shared/types.js';
 import { AppError, FORBIDDEN_WORKSPACE_PATHS, normalizeProjectPath } from '@/shared/utils.js';
 
-const IGNORED_DIRECTORY_NAMES = new Set([
+/**
+ * Directory names hidden from the tree until a user stores their own list.
+ * Every entry is generated output that a project can rebuild, so hiding it by
+ * default keeps large workspaces inside the entry budget below.
+ */
+export const DEFAULT_IGNORED_DIRECTORY_NAMES: readonly string[] = [
```
[the remark this answers is not in the document]
Comment (HandyS11) [reply server/modules/file-tree/file-tree.service.ts]: Agreed, done in 66ef360 — the constant is module-private again. The two tests that referenced it now assert the behaviour instead of the identity: the fallback case asserts `ignoredDirectories` deep-equals `defaults` (which is the actual contract) plus the presence of `bin`/`obj`, and the stored-list case asserts against `getIgnoredDirectories().defaults`.

File: src/components/settings/view/tabs/FilesSettingsTab.tsx [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+
+          <div className="flex gap-2">
+            <Input
+              value={newDirectory}
+              onChange={(event) => setNewDirectory(event.target.value)}
+              onKeyDown={(event) => {
+                if (event.key === 'Enter') {
+                  event.preventDefault();
+                  addDirectory();
+                }
```
[the remark this answers is not in the document]
Comment (HandyS11) [reply src/components/settings/view/tabs/FilesSettingsTab.tsx:165]: Fixed in 66ef360, following the existing guard in `useChatComposerState.ts:1148`: the handler returns when `event.nativeEvent.isComposing` is set before it reaches `addDirectory()`. Verified in a browser — dispatching Enter with `isComposing: true` leaves the chip count at 24, while a normal Enter adds the entry (25).

**Comment by HandyS11** [comment]
Review feedback addressed in 66ef360 — all five findings from Copilot and CodeRabbit, replies on each thread.

- **Save could wipe the config after a failed load** (Copilot, the one real bug): the tab now renders only the error until it has actually read the current settings, so no Save or Reset button exists to overwrite the stored list with the empty one the component starts with.
- **Enter during IME composition** (CodeRabbit): guarded via `event.nativeEvent.isComposing`, matching `useChatComposerState.ts:1148`.
- **Orphaned JSDoc block** (Copilot): the gateway type moved above the `FileTreeServiceDependencies` doc so each comment is attached to its own type.
- **`DEFAULT_IGNORED_DIRECTORY_NAMES` exported only for tests** (Copilot): now module-private; the tests assert the contract (`ignoredDirectories` deep-equals `defaults` when nothing is stored) rather than the constant's identity.
- **Missing empty-list coverage** (CodeRabbit): added a test pinning the `null` vs `[]` distinction — an explicitly empty list hides nothing and leaves `bin` visible. I checked it has teeth by making the resolver treat empty as unset; the test fails as expected.

`npm test` 273 pass, `typecheck`/`lint`/`build` clean. Re-verified in a running instance: forcing the GET to 500 shows the error with 0 Save/Reset buttons, a composing Enter leaves the chip count unchanged while a plain Enter adds the entry, and the default list still loads and saves normally.
